Source/WTF/wtf/RetainPtr.h

```diff
@@ -325,15 +319,18 @@ template<typename T> inline RetainPtr<typename RetainPtr<T>::HelperPtrType> adop
 {
     static_assert(std::is_convertible_v<T, id>, "Don't use adoptNS with Core Foundation pointer types, use adoptCF.");
 #if __has_feature(objc_arc)
-    return ptr;
-#elif defined(OBJC_NO_GC)
-    using ReturnType = RetainPtr<typename RetainPtr<T>::HelperPtrType>;
-    return ReturnType { ptr, ReturnType::Adopt };
-#else
-    RetainPtr<typename RetainPtr<T>::HelperPtrType> result = ptr;
-    [ptr release];
-    return result;
+    // When ARC is enabled, NS_RELEASES_ARGUMENT emits a release into the
+    // callee. Since we're going to hold on to the pointer, we need to balance
+    // that release with a matching retain.
+
+    // In most ARC code, this all happens automatically when you store to an
+    // ObjC property. We're storing to a C++ object, so we need to do this step
+    // manually.
+
+    // Perhaps surprisingly, the net effect is an adopt, with zero refcount churn.
+    ptr = (__bridge T)(__bridge_retained CFTypeRef)ptr;
```

Comment:
Is there a way to write a test for ref churn here? e.g. by overriding `retain` and `release` selectors?

Reply:
I don't think so, because it's illegal to override retain and release when ARC is enabled.

Reply:
I see. That's unfortunate.